From d5c1bb71f57ad7459d4aa23c84681e8356faf8c9 Mon Sep 17 00:00:00 2001
From: Valentin Catalin Neacsu <valentin-catalin.neacsu@nxp.com>
Date: Thu, 6 Dec 2018 14:28:57 +0200
Subject: [PATCH 735/767] net: phy: aquantia: Read link status twice

Read link status twice because register is latching low.

Signed-off-by: Valentin Catalin Neacsu <valentin-catalin.neacsu@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.12.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/net/phy/aquantia.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/phy/aquantia.c b/drivers/net/phy/aquantia.c
index 13533c032e8f..e0b13d91f959 100644
--- a/drivers/net/phy/aquantia.c
+++ b/drivers/net/phy/aquantia.c
@@ -309,7 +309,10 @@ static int aquantia_read_status(struct phy_device *phydev)
 {
 	int reg;
 
+	/* Read the link status twice; the bit is latching low */
 	reg = aquantia_read_reg(phydev, MDIO_MMD_AN, MDIO_STAT1);
+	reg = aquantia_read_reg(phydev, MDIO_MMD_AN, MDIO_STAT1);
+
 	if (reg & MDIO_STAT1_LSTATUS)
 		phydev->link = 1;
 	else
-- 
2.17.0

